include/zephyr/rtio/rtio.h

```diff
@@ -1112,6 +1112,57 @@ static inline struct rtio_sqe *rtio_sqe_acquire(struct rtio *r)
 	return &iodev_sqe->sqe;
 }
 
+/**
+ * @brief Acquire a number of submission queue events if available
+ *
+ * @warn The sqe array is in an undefined state if the return value is -ENOMEM
+ *
+ * @param r RTIO context
+ * @param n Number of submission queue events to acquire
+ * @param sqes A pointer to an array of rtio_sqe struct pointers of size n
+ *
+ * @retval 0 success
+ * @retval -ENOMEM out of memory
+ */
+static inline int rtio_sqe_acquire_n(struct rtio *r, size_t n, struct rtio_sqe **sqes)
+{
+	size_t i;
+
+	for (i = 0; i < n; i++) {
+		struct rtio_iodev_sqe *iodev_sqe = rtio_sqe_pool_alloc(r->sqe_pool);
```

Comment:
Nit: we can move the `iodev_sqe` declaration to the beginning of the function, and re-use the pointer throughout the function.

```c
static inline int rtio_sqe_acquire_n(struct rtio *r, size_t n, struct rtio_sqe **sqes)
{
	struct rtio_iodev_sqe *iodev_sqe;
	size_t i;

	for (i = 0; i < n; i++) {
		iodev_sqe = rtio_sqe_pool_alloc(r->sqe_pool);
	
	// ...
}